### ISSUE TITLE
SVG output is broken

### ISSUE THREAD

**Issue by rlafuente** [shoebot/shoebot#179]
I might have broken this while taking a stab at Jupyter.

Right now, when generating an SVG file, Shoebot hangs when finishing the surface (via `surface.finish()`).

This is a big blocker, and I'll be trying to fix this before anything else.

**Comment by stuaxo** [comment]
No probs, I can try and fix snapshot later when I'm not at work - but it would be good if I could fixup snapshot on the pre-jupyterised version.

Can you make a new branch e.g. for the jupyter support "jupyter" - and then revert master back the pre-changed version of snapshot ?

This way it'll be easier to isolate things.   I'll try and actually use branches for dev from now as well :)

**Comment by stuaxo** [comment]
It turns out we have a unit test ```test_file_output.py```, and that seems to create images OK before the patch

You can comment out the unlink() that deletes the file, and add something to print the filename.

**Comment by stuaxo** [comment]
Though it looks like tests were passing according to Travis.. I'll look into it

**Comment by rlafuente** [comment]
Yes, I'll revert now. In hindsight it's obvious I should have made a branch, sorry for the hassle this caused.

**Comment by rlafuente** [comment]
Hmm, even after reverting, the weird behaviour is still there. I'm testing it with `sbot examples/grid/balls.bot -o test.svg`, which hangs and has to be killed with `Ctrl-\`

**Comment by stuaxo** [comment]
Ah, that's good, it'll be easier to fix :)

**Comment by stuaxo** [comment]
Is this on Linux?

**Comment by stuaxo** [comment]
Re your earlier comment - to be fair I've been developing on master for ages, it's a little cowboy like - but I was the only one for a while.

**Comment by stuaxo** [comment]
Definitely a weird bug, can you use png in the meantime? 

I will try and use git bisect to find out whats up.

**Comment by stuaxo** [comment]
I couldn't get earlier versions working either, so there might be something going on with cairo - seems like it will take more fixing :/

**Comment by rlafuente** [comment]
I can definitely live with PNG for the time being, no rush on my side! It's just that it's a pretty common output format that we'll want to support fully :-)

The bug popped up while trying to add SVG support to Jupyter. I'm not sure for how long it might have been lurking, but probably not that long.

FWIW, this worked without issue:

```python
import shoebot
import cairocffi as cairo

surface = cairo.SVGSurface('example.svg', 200, 200)
ctx = cairo.Context(surface)
bot = shoebot.create_bot(surface)

bot.size(200, 200)
bot.background(0.8)
bot.fill(0.2)
bot.ellipse(10, 10, 100, 100)
# clunky call, but that's the only way I could do this
bot._canvas._drawqueue.render(ctx)
bot.finish()
```

But not with `sbot examples/grid/balls.bot -o example.svg`.

**Comment by stuaxo** [comment]
The difference between balls.bot and hypnoval.bot is the second one is "dynamic" (has a draw function).
So maybe it's broken for non dynamic bots.

Maybe we could drop the

**Comment by stuaxo** [comment]
(oops fatfingered this + closed it by accident)

**Comment by stuaxo** [comment]
Need to retest this with the recent graphics changes (I'm expecting it is still broken though)

**Comment by stuaxo** [comment]
We need some tests that run shoebot end to end with `-o`  - ideally it would kill the process if it takes too long too.

**Comment by stuaxo** [comment]
The first issue I found is that "windowed" is being set even though we are outputting to SVG.

The second issue of the freeze is a lot weirder.

We freeze in cairo_sink.rendering_finished() at `surface.finish()`:
https://github.com/shoebot/shoebot/blob/master/shoebot/core/cairo_sink.py#L104

If the `finish()`, and `flush()` calls are removed then the freeze happens on return from the function - implying something getting triggered in GC being the culprit.
